modules/widget-visibility/widget-conditions.php

```diff
@@ -679,12 +693,39 @@ public static function filter_widget( $instance ) {
 				}
 			}
 
-			if ( $condition_result )
+			if (
+				isset( $instance['conditions']['match_all'] )
+				&& $instance['conditions']['match_all'] == '1'
+				&& ! $condition_result
+			) {
+
+				// In case the match_all flag was set we quit on first failed condition
+				break;
+			} elseif(
+				(
+					! isset( $instance['conditions']['match_all'] )
+					|| $instance['conditions']['match_all'] !== '1'
+				)
+				&& $condition_result
+			) {
+
+				// Only quit on first condition if the match_all flag was not set
 				break;
+			}
 		}
 
-		if ( ( 'show' == $instance['conditions']['action'] && ! $condition_result ) || ( 'hide' == $instance['conditions']['action'] && $condition_result ) )
+		if (
+			(
+				'show' == $instance['conditions']['action']
+				&& ! $condition_result
+			)
+			|| (
```

Comment:
I personally find broken out conditionals like this to be harder to read, but if we do, can we please do it as `) || (` instead?